skimage/segmentation/random_walker_segmentation.py

```diff
@@ -208,16 +208,19 @@ def _solve_linear_system(lap_sparse, B, tol, mode):
                 )
             M = None
         elif mode == 'cg_j':
-            M = sparse.diags(1.0 / lap_sparse.diagonal())
+            n = lap_sparse.shape[-1]
+            M = sparse.dia_array((1.0 / lap_sparse.diagonal(), 0), shape=(n, n))
         else:
             # mode == 'cg_mg'
             lap_sparse = lap_sparse.tocsr()
+            lap_sparse.indptr = lap_sparse.indptr.astype('int32')
```

Comment:
Should we not check here whether this cast is valid, considering the values in `indptr`?

Reply:
@dschult Thoughts?